lib/node_modules/@stdlib/types/index.d.ts

```diff
@@ -593,6 +593,37 @@ declare module '@stdlib/types/array' {
 	* const x: Collection<number> = [ 1, 2, 3 ];
 	*/
 	type Collection<T = any> = Array<T> | TypedArray | ArrayLike<T>; // eslint-disable-line @typescript-eslint/no-explicit-any
+
+	/**
+	* Type mapping for the following array data types:
+	*
+	* -   `float64`: double-precision floating-point numbers (IEEE 754)
+	* -   `float32`: single-precision floating-point numbers (IEEE 754)
+	* -   `complex128`: double-precision complex floating-point numbers
+	* -   `complex64`: single-precision complex floating-point numbers
+	* -   `int32`: 32-bit two's complement signed integers
+	* -   `uint32`: 32-bit unsigned integers
+	* -   `int16`: 16-bit two's complement signed integers
+	* -   `uint16`: 16-bit unsigned integers
+	* -   `int8`: 8-bit two's complement signed integers
+	* -   `uint8`: 8-bit unsigned integers
+	* -   `uint8c`: 8-bit unsigned integers clamped to `0-255`
+	* -   `generic`: generic JavaScript values
+	*/
+	type ArrayDataTypeMap<T> = { // eslint-disable-line @typescript-eslint/consistent-type-definitions
```

Comment:
What is the typical naming convention for something like this?

Reply:
It's commonly referred to as a "type map"; canonical example I have been coming across is mapping HTML tag names to their respective HTML elements.  

Notice that I added various other subsets of this during refactoring in `types/index.d.ts`. Didn't include the `DataTypeMap` suffix everywhere as it would have led to some rather long names, e.g. `RealFloatingPointArrayDataTypeMap`, but maybe we should? You have a preference?

Reply:
I think we can drop the `Array` prefix and do `DataTypeMap`, `RealFloatingPointDataTypeMap`, etc. The "Array" portion is implied given that these will be defined in the `array` module.

Reply:
Done.